### ISSUE TITLE
Update all playground examples to use AiModelBuilder facade pattern

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#781]
## Overview

Currently only **26 of 160 playground examples (16%)** use the `AiModelBuilder` facade pattern. All examples should use this pattern to:
- Demonstrate the recommended API usage
- Hide implementation complexity
- Provide consistent user experience

## Available AiModelBuilder Configuration Methods

The following methods exist and should be used where applicable:

| Method | Use Case |
|--------|----------|
| `ConfigureDataLoader` | All supervised learning |
| `ConfigureModel` | ML models implementing `IFullModel` |
| `ConfigurePreprocessing` | Scalers, transformers |
| `ConfigureOutlierRemoval` | Anomaly detection |
| `ConfigureAutoML` | AutoML model selection |
| `ConfigureLoRA` | LoRA fine-tuning |
| `ConfigureFineTuning` | General fine-tuning |
| `ConfigureReinforcementLearning` | RL agents |
| `ConfigureCrossValidation` | Cross-validation |
| `ConfigureMetaLearning` | MAML, prototypical networks |
| `ConfigureDistributedTraining` | Federated, data parallel |
| `ConfigureBiasDetector` | Bias detection |
| `ConfigureQuantization` | Model quantization |
| `ConfigureAugmentation` | Data augmentation |
| `ConfigureHyperparameterOptimizer` | Hyperparameter tuning |
| `ConfigureExperimentTracker` | Experiment tracking |
| `ConfigurePromptTemplate` | Prompt engineering |

### Category 1: ML Models (37) - Use ConfigureModel + ConfigureDataLoader

These models should implement `IFullModel` and use the standard pattern:

```csharp
var loader = DataLoaders.FromArrays(features, labels);
var result = await new AiModelBuilder<double, Matrix<double>, Vector<double>>()
    .ConfigureDataLoader(loader)
    .ConfigureModel(new SomeModel<double>(options))
    .BuildAsync();
```

- [ ] simple-regression
- [ ] ridge-regression-basic
- [ ] polynomial-regression
- [ ] lasso-regression
- [ ] elastic-net
- [ ] kmeans
- [ ] dbscan
- [ ] hdbscan
- [ ] fuzzy-cmeans
- [ ] meanshift
- [ ] spectral-clustering
- [ ] kmeans-advanced
- [ ] neural-net-basic
- [ ] neural-net-regression
- [ ] arima
- [ ] exponential-smoothing
- [ ] image-classification-cnn
- [ ] resnet-transfer
- [ ] object-detection-yolo
- [ ] tfidf-vectorizer
- [ ] text-classification-nn
- [ ] named-entity-recognition
- [ ] gp-regression
- [ ] voting-ensemble
- [ ] stacking
- [ ] gcn
- [ ] gat
- [ ] collaborative-filtering
- [ ] content-based
- [ ] bayesian-nn
- [ ] rbf-network
- [ ] video-classification
- [ ] nerf-basics
- [ ] point-cloud-processing
- [ ] geometric-transforms
- [ ] vector-store
- [ ] semantic-search

### Category 2: Fine-Tuning (4) - Use ConfigureLoRA/ConfigureFineTuning

- [ ] lora-basic
- [ ] qlora
- [ ] dora
- [ ] adalora

### Category 3: Preprocessing (9) - Use ConfigurePreprocessing

- [ ] standard-scaler
- [ ] minmax-scaler
- [ ] preprocessing-pipeline
- [ ] variance-threshold
- [ ] correlation-selection
- [ ] smote
- [ ] image-augmentation
- [ ] text-augmentation
- [ ] feature-extraction

### Category 4: Advanced ML (24) - Various Configure methods

- [ ] hyperparameter-tuning (ConfigureHyperparameterOptimizer)
- [ ] bayesian-optimization (ConfigureAutoML with BayesianOptimization)
- [ ] kfold-cv (ConfigureCrossValidation)
- [ ] stratified-kfold (ConfigureCrossValidation)
- [ ] maml (ConfigureMetaLearning)
- [ ] prototypical-networks (ConfigureMetaLearning)
- [ ] teacher-student (ConfigureKnowledgeDistillation)
- [ ] ddpm-basics
- [ ] domain-adaptation
- [ ] conformal-prediction
- [ ] pinn-basics
- [ ] pruning (ConfigureCompression)
- [ ] quantization (ConfigureQuantization)
- [ ] uncertainty-sampling
- [ ] fedavg (ConfigureDistributedTraining)
- [ ] ewc
- [ ] contrastive (ConfigureSelfSupervisedLearning)
- [ ] causal-ml
- [ ] bayesian-inference
- [ ] neuro-symbolic
- [ ] simple-ga
- [ ] nsga-ii
- [ ] grid-search (ConfigureHyperparameterOptimizer)
- [ ] bayesian-opt (ConfigureHyperparameterOptimizer)

### Category 5: Interpretability (3) - Use ConfigureBiasDetector/etc

- [ ] lime-explanation
- [ ] feature-importance
- [ ] bias-detection (ConfigureBiasDetector)

### Category 6: Infrastructure (12) - Various methods

- [ ] onnx-export (ConfigureExport)
- [ ] model-save-load
- [ ] tracking-basics (ConfigureExperimentTracker)
- [ ] rest-api
- [ ] classification-metrics
- [ ] regression-metrics
- [ ] fp16-training (ConfigureMixedPrecision)
- [ ] autodiff-basics
- [ ] data-parallel (ConfigureDistributedTraining)
- [ ] prompt-templates (ConfigurePromptTemplate)
- [ ] self-attention
- [ ] multi-head-attention

### Category 7: Agents (2)

- [ ] react-agent (ConfigureAgentAssistance or ConfigureReasoning)
- [ ] plan-execute-agent

### Category 8: Component Demos (23)

These demonstrate individual components. Consider showing them within AiModelBuilder context:

- [ ] mse-loss, cross-entropy-loss, huber-loss, focal-loss
- [ ] relu-activation, gelu-activation, sigmoid-softmax, swish-mish
- [ ] step-lr-scheduler, cosine-annealing-scheduler, one-cycle-scheduler
- [ ] l1-regularization, l2-regularization, elastic-net-regularization
- [ ] gaussian-kernel, polynomial-kernel, matern-kernel
- [ ] minmax-normalizer, zscore-normalizer, robust-scaler
- [ ] adam-optimizer, sgd-momentum, adamw-optimizer

### Category 9: Math/Utility (20)

Educational examples - may need different approach:

- [ ] hello-world
- [ ] tensor-basics, matrix-operations, matrix-multiplication, vector-operations
- [ ] euclidean-distance, manhattan-distance, cosine-distance, minkowski-distance, chebyshev-distance
- [ ] basic-statistics, correlation, mse-rmse-mae
- [ ] svd-decomposition, eigenvalue-decomposition, linear-interpolation
- [ ] window-functions, bpe, dwt, silhouette-score

## Automated Testing Required

Need CI/CD integration to verify:
1. All examples compile
2. All examples execute without runtime errors
3. All examples produce meaningful output (not null/empty)

## Acceptance Criteria

- [ ] All 160 examples use AiModelBuilder facade pattern
- [ ] Automated tests verify examples work
- [ ] CI/CD runs example tests on every PR
